### PR TITLE
documentation - markdown formatting

### PR DESCRIPTION
**Pull request by lb-**
- apply general fixes to existing markdown documentation
- various cases of rst syntax still used
- update some links to be the new format
- clean up line breaks (prettier)
- [x] Does the code comply with the style guide?

### CODE REVIEW

File: docs/topics/pages.md [hunk 1]
```diff
@@ -335,18 +337,19 @@ Wagtail can nest the content of other models within the page. This is useful for
...
+-   It must have a `ParentalKey` to the parent model
 
-```{note} django-modelcluster and ParentalKey
+````{note} django-modelcluster and ParentalKey
 The model inlining feature is provided by [django-modelcluster](https://github.com/torchbox/django-modelcluster) and the `ParentalKey` field type must be imported from there:
 
 ```python
     from modelcluster.fields import ParentalKey
-```
+````
```
Comment (zerolab) [inline docs/topics/pages.md:348]: this doc seems to get extra backticks for this code block and below the `ParentalKey` line

**Review by zerolab — CHANGES_REQUESTED** [review]

File: docs/topics/pages.md (hunk 1 above)
Comment (lb-) [reply docs/topics/pages.md:348]: Hmm ok I'll have to run this locally to check. Probably an issue with the code snippet within the admonition.

Comment (lb-) [reply docs/topics/pages.md]: should be fixed now, the actual rendering was working better than the current production docs but I have fixed up the odd extra backtick.

Note: We need four backticks on the outer admonition due to there being three for the inner code snippet.

Here is the validation screenshot (with the newly pushed changes), left is current stable, right is local docs.
Screen Shot 2022-06-01 at 8 40 52 pm
See that the text 'For example, the following inline model can be used to add related links (a list of name, url pairs) to the `BlogPage` model:' is incorrectly showing inside the code block in the current stable docs.

**Comment by lb-** [comment]
@zerolab requesting another review of this if possible.

File: docs/advanced_topics/api/v2/usage.md [hunk 2]
```diff
@@ -337,6 +337,7 @@ to only include descendants of that page (children, grandchildren etc.).
 ### Filtering pages by site
 
 ```{versionadded} 4.0
+
```
Comment (zerolab) [inline docs/advanced_topics/api/v2/usage.md:340]: this bit feels a tad odd, but not a big issue

**Review by zerolab — APPROVED** [review]
@lb- nice work.

There's a small change around myst wrappers for RST that looks odd to me, but it is inocuous enough.

Will the next step be to enforce the markdown formatting during linting?

File: docs/advanced_topics/api/v2/usage.md (hunk 2 above)
Comment (lb-) [reply docs/advanced_topics/api/v2/usage.md:340]: Yeah. I tested both outputs - with or without the spacing and it makes no visible difference. So I just went with what prettier did.

**Comment by lb-** [comment]
> @lb- nice work.
> 
> There's a small change around myst wrappers for RST that looks odd to me, but it is inocuous enough.
> 
> Will the next step be to enforce the markdown formatting during linting?

I'll raise a new issue to see if we want to enforce prettier in the docs for some sections (maybe not release notes for example). 

We can discuss pros/cons there.

**Comment by lb-** [comment]
Thanks for the approval @zerolab  - I will merge this in.

Also - discussion raised about whether we should enforce prettier for markdown files - please add your thoughts there. https://github.com/wagtail/wagtail/discussions/8753

**Pull request merged**